src/components/SortDropdown.jsx

```diff
@@ -0,0 +1,49 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { makeStyles } from '@material-ui/core/styles';
+import InputLabel from '@material-ui/core/InputLabel';
+import MenuItem from '@material-ui/core/MenuItem';
+import FormControl from '@material-ui/core/FormControl';
+import Select from '@material-ui/core/Select';
+
+const useStyles = makeStyles((theme) => ({
+  formControl: {
+    margin: theme.spacing(1),
+    minWidth: 120,
+  },
+}));
+
+function SortDropdown(props) {
+  const { items, updateSorting } = props;
+  const label = 'Sorting';
+  const [choice, setChoice] = useState('deadlineSoon');
+  const classes = useStyles();
+
+  function selectChoice(event) {
+    console.log(`switch Value changed to: ${event.target.value}`);
+    setChoice(event.target.value);
+    updateSorting(event.target.value);
+  }
+
+  return (
+    <>
+      <FormControl variant="outlined" className={classes.formControl}>
+        <InputLabel>{label}</InputLabel>
+        <Select value={choice} onChange={selectChoice} label={label}>
+          {Object.keys(items).map((key) => (
+            <MenuItem key={key} value={key}>
+              {items[key]}
+            </MenuItem>
+          ))}
+        </Select>
+      </FormControl>
```

Comment:
This looks pretty much like `FilterDropdown`. When you see this sort of pattern you need to ask yourself if you can reuse the existing component. If the answer is no (which is why I assume you created this component), then you need to ask yourself if the functionality you need is specific to the sorting dropdown or if they're generally applicable to filter dropdowns in general. In this case, the questions to ask are:
- Is passing a function to invoke on selection change relevant for filter dropdowns in general?
- Is passing an initial selection/choice relevant for filter dropdowns in general?
- Is passing a list of items/labels apart from list a list of values relevant for filter dropdowns in general?

The answer to all those questions is yes. Since the functionality you need is relevant for filter dropdowns in general, it therefore makes sense to reuse `FilterDropdown` and add `onChange`, `initialChoice`, and `values` props to instead. What you'll notice then is that this class basically becomes a thin wrapper around `FilterDropdown` that's probably unnecessary. At least for now. Much later on if we have similar sort dropdowns in other places then it'll make sense to refactor a reusable sort dropdown component.

Reply:
Essentially you want the FilterDropdown to contain the logic for sorting and then also you want the FilterDropdown to handle the change in values passed to the Scholarships.list not how it currently is in the FilterBar. BTW I had trouble passing down function that was passed via props, do I have to make a new function within that class that calls the props function and then pass the created function that make the call to the prop function to the FilterDropdown or something?

Reply:
No, it wouldn't have _sorting_ specific logic, it'd have _filter dropdown_ specific logic. That is, logic that one would expect all filter dropdowns can do.

 Basically you'd replace this code with something like this:
```jsx
<FilterDropDown
          label="Sorting"
          items={sortingOptions} // "Deadline (earliest first)", etc.
          values={sortingValues}  // "deadlineSoon", etc.
          defaultValue="deadlineSoon"
          onChange={updateSorting}  // function that updates sortField/sortDir based on the value selected
        />
```
Notice how the _sorting_ specific data/logic is passed in as general filter dropdown properties. You're not coding _sorting_ specific logic in `FilterDropDown`. You're just telling `FilterDropDown` what values to use to style itself and how it should react on a selection change.

Reply:
To pass a function that was passed via props just pass it. E.g. in `FilterBar` you can do this

```
<SortDropdown setSortDir={setSortDir} setSortField={setSortField}>
```
where `setSortDir` and `setSortField` are properties in both the `FilterBar` and `SortDropdown` components. Then inside `SortDropdown` your `updateSorting` function does the sorting-specific logic you currently have in `FilterBar`.

Reply:
Sounds good changes are pushed